addons/binding/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/handler/ExecHandler.java

```diff
@@ -53,191 +58,305 @@
                     }
-                    isr.close();
-                } catch (IOException e) {
-                    logger.error("An exception occurred while reading the stderr when executing '{}' : '{}'",
-                            new Object[] { commandLine.toString(), e.getMessage() });
-                }
 
-                boolean exitVal = false;
-                try {
-                    exitVal = proc.waitFor(timeOut, TimeUnit.MILLISECONDS);
-                } catch (InterruptedException e) {
-                    logger.error("An exception occurred while waiting for the process ('{}') to finish : '{}'",
-                            new Object[] { commandLine.toString(), e.getMessage() });
-                }
+                    if (!hasExited) {
+                        logger.warn("Forcibly termininating the process ('{}') after a timeout of {} seconds",
+                                commandLine, config.getTimeout().intValue());
+                        proc.destroyForcibly();
+                    }
 
-                if (!exitVal) {
-                    logger.warn("Forcibly termininating the process ('{}') after a timeout of {} ms",
-                            new Object[] { commandLine.toString(), timeOut });
-                    proc.destroyForcibly();
-                }
+                    updateState(RUN, OnOffType.OFF);
+                    updateState(EXIT, new DecimalType(proc.exitValue()));
+
+                    outputBuilder.append(errorBuilder.toString());
 
-                updateState(RUN, OnOffType.OFF);
-                updateState(EXIT, new DecimalType(proc.exitValue()));
+                    String transformedResponse = StringUtils.chomp(outputBuilder.toString());
+                    String transformation = config.getTransform();
 
-                outputBuilder.append(errorBuilder.toString());
+                    if (StringUtils.isNotBlank(transformation)) {
+                        transformedResponse = transformString(transformedResponse, transformation);
+                    }
 
-                outputBuilder.append(errorBuilder.toString());
+                    updateState(OUTPUT, new StringType(transformedResponse));
 
-                String transformedResponse = StringUtils.chomp(outputBuilder.toString());
-                String transformation = (String) getConfig().get(TRANSFORM);
+                    DateTimeType stampType = new DateTimeType(Calendar.getInstance());
+                    updateState(LAST_EXECUTION, stampType);
 
-                if (transformation != null && transformation.length() > 0) {
-                    transformedResponse = transformResponse(transformedResponse, transformation);
                 }
 
-                updateState(OUTPUT, new StringType(transformedResponse));
-
-                DateTimeType stampType = new DateTimeType(Calendar.getInstance());
-                updateState(LAST_EXECUTION, stampType);
+            } catch (
```

Comment:
This formatting looks really bizar can you maybe reformat all the code for this binding using the default eclipse setup?